TACC/Core-Styles
# Fix/tup 660 form badge font size zero can remove all space

## Overview

Do **not** remove space between a label and a form badge.

<details><summary>Why?</summary>

Because the Django plugin does not add whitespace, so the result is that "(required)" touches the label text. It is better to have too much space than not enough space.

</details>

## Related

- [TUP-660](https://jira.tacc.utexas.edu/browse/TUP-660)

## Changes

- **increased** `font-size` for `:--form__badge`
- **added** comment referencing this PR

## Testing

1. Open http://localhost:3002/components/detail/c-form--portal.
2. Edit markup so there is **no** _markup_ whitespace between label text and `c-form__star`.
3. Verify there is **still** _visible_ space between label text and "(required)".

## UI

| before WITH whitespace | before SANS whitespace | after WITH whitespace | after SANS whitespace |
| - | - | - | - |
| ![before WITH whitespace](https://github.com/TACC/Core-Styles/assets/62723358/f1ef2437-9787-4a0d-9115-f5ddc70f6f54) | ![before SANS whitespace](https://github.com/TACC/Core-Styles/assets/62723358/b16c3549-8ee8-4c33-a2ad-15bb51afef9c) | ![after WITH whitespace](https://github.com/TACC/Core-Styles/assets/62723358/66e03913-de1c-4908-bdf3-18aeb0e73bec) | ![after SANS whitespace](https://github.com/TACC/Core-Styles/assets/62723358/7cc5754b-b08c-4add-b59b-e2fbdfc1bb40) |

## Notes

Alternate **future**[^1] solution is to **set** [`white-space-collapse: collapse`](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space-collapse) and **remove** `font-size: 0` and **add** `margin-left: 0.5em` for **all** form badges.

[^1]: As of 2023-11-21, browser support is inadequate.
